### PR TITLE
MMF-999 Add management of quality gates on organizations

### PR DESCRIPTION
**Pull request by ehartmann**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QualityGates.java#L97](https://github.com/SonarSource/sonarqube/blob/387bb4166561d6715f3ee3a6f54c68a1c342c32f/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L97): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityGates.java#L112](https://github.com/SonarSource/sonarqube/blob/387bb4166561d6715f3ee3a6f54c68a1c342c32f/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L112): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v70/PopulateUuidOnQualityGates.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("select id, name from quality_gates where uuid is null");
+    massUpdate.rowPluralName("quality gates");
+    massUpdate.update("update quality_gates set uuid=?, updated_at=? where id=?");
+    massUpdate.execute((row, update) -> {
+      String name = row.getString(2);
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v70/PopulateUuidOnQualityGates.java]: MINOR Remove this unused "name" local variable. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1481)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QualityGates.java#L97](https://github.com/SonarSource/sonarqube/blob/0c3b8c3955617b2ae55f61081364b5ecda09653d/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L97): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityGates.java#L112](https://github.com/SonarSource/sonarqube/blob/0c3b8c3955617b2ae55f61081364b5ecda09653d/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L112): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-db-dao/src/main/java/org/sonar/db/qualitygate/QGateWithOrgDto.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  @Override
+  public QGateWithOrgDto setBuiltIn(boolean builtIn) {
+    super.setBuiltIn(builtIn);
+    return this;
+  }
+
+  @Override
+  public QualityGateDto setCreatedAt(Date createdAt) {
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualitygate/QGateWithOrgDto.java]: MINOR Remove this method to simply inherit it. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1185)

File: server/sonar-db-dao/src/main/java/org/sonar/db/qualitygate/QGateWithOrgDto.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    return super.setCreatedAt(createdAt);
+  }
+
+  @Override
+  public QualityGateDto setUpdatedAt(Date updatedAt) {
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualitygate/QGateWithOrgDto.java]: MINOR Remove this method to simply inherit it. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1185)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QualityGates.java#L97](https://github.com/SonarSource/sonarqube/blob/989986142d38f65ad0392fd0c7ef49c2f0cc1cd7/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L97): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityGates.java#L112](https://github.com/SonarSource/sonarqube/blob/989986142d38f65ad0392fd0c7ef49c2f0cc1cd7/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L112): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QualityGates.java#L97](https://github.com/SonarSource/sonarqube/blob/1bbb957824a23e93e245e34d7451866e414a37c2/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L97): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityGates.java#L112](https://github.com/SonarSource/sonarqube/blob/1bbb957824a23e93e245e34d7451866e414a37c2/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L112): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QualityGates.java#L97](https://github.com/SonarSource/sonarqube/blob/7ac7c780925706ab81b64d6078642f66cdab9390/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L97): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityGates.java#L112](https://github.com/SonarSource/sonarqube/blob/7ac7c780925706ab81b64d6078642f66cdab9390/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L112): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QualityGates.java#L97](https://github.com/SonarSource/sonarqube/blob/e071c1ce447467dd7c8b19e6c569b6663ec82ae9/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L97): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QualityGates.java#L112](https://github.com/SonarSource/sonarqube/blob/e071c1ce447467dd7c8b19e6c569b6663ec82ae9/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java#L112): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by ehartmann** [comment]
Try reopening

**Pull request merged**